### DISCUSSION TITLE
Units of dimensionless quantities

### DISCUSSION THREAD

**Discussion by climbfuji** [ESCOMP/CCPPStandardNames#18] [Q&A]
The dictionary of standard names has a unit of `1` for dimensionless quantities, which is scientifically correct.

The existing metadata in ccpp-physics and associated host models (UFS, SCM) uses `none` for dimensionless quantities.

It is of course possible to write a unit conversion code that just returns the identity if the code generator encounters `1` and `none` between host model and physics, but this may still create overhead by calling a function that does nothing (although chances are that the compiler optimizes it out).

Alternatively, we could add logic to the code generator to ignore such dummy unit conversions.

Best of course would be to agree to either `1` or `none`.

Looking forward to hearing your thoughts.

**Comment by dudhia** [comment]
Often there is a helpful choice like kg kg-1 or mol mol-1 and that should be done as much as possible.
For truly non-dimensional things it may be case by case, but I would need to see them to know which is better. For example, if they really are numbers, 1 would be a good unit, but maybe they are indices(?) is there an index dimension? Other examples not in these categories?

**Reply by climbfuji** [reply]
These are good points. Two variables that I came across yesterday as examples:
```
integer :: ccpp_constant_one = 1
```
As you said, `unit=1` would be a good choice here. `capgen` has `1` as units, in `ccpp_prebuild` that variable didn't exist.

The other one was
```
character(len=512) :: errmsg = 'foo'
```
which in `capgen` also has `1` as units, but `none` in `ccpp_prebuild`.

**Reply by grantfirl** [reply]
This was discussed A LOT several years ago and we arrived at 'none'. Dave Gill was in on those conversations and spoke for NCAR models at the time. I agree that kg kg-1 etc. should be used where appropriate, but otherwise I think that the default dimensionless unit should remain 'none' as agreed originally.

**Comment by dudhia** [comment]
Yes, I see other units like count, index, flag, frac, so at least they are
being distinguished.
Consistency is what is needed and I have no problem with 'none'.
However I note that some parameters for schemes are labeled 'none' and I am
skeptical of that,
but it would take some digging to correct those.
